src/k8s/pkg/k8sd/setup/certificates.go

```diff
@@ -11,54 +11,73 @@ import (
 )
 
 // ensureFile creates fname with the specified contents, mode and owner bits.
-// ensureFile will delete the file if contents is an empty string.
-func ensureFile(fname string, contents string, uid, gid int, mode fs.FileMode) error {
+// It will delete the file if contents is an empty string.
+// It returns true if a file was updated (or deleted) and any error that occured.
+func ensureFile(fname string, contents string, uid, gid int, mode fs.FileMode) (bool, error) {
 	if contents == "" {
-		if err := os.Remove(fname); err != nil && !os.IsNotExist(err) {
-			return fmt.Errorf("failed to delete: %w", err)
+		if err := os.Remove(fname); err != nil {
+			return false, fmt.Errorf("failed to delete: %w", err)
 		}
-		return nil
+
+		return true, nil
 	}
 
 	if err := os.WriteFile(fname, []byte(contents), mode); err != nil {
-		return fmt.Errorf("failed to write: %w", err)
+		return false, fmt.Errorf("failed to write: %w", err)
 	}
 	if err := os.Chown(fname, uid, gid); err != nil {
-		return fmt.Errorf("failed to chown: %w", err)
+		return false, fmt.Errorf("failed to chown: %w", err)
 	}
 	if err := os.Chmod(fname, mode); err != nil {
-		return fmt.Errorf("failed to chmod: %w", err)
+		return false, fmt.Errorf("failed to chmod: %w", err)
 	}
 
-	return nil
+	return true, nil
```

Comment:
this does not look right to me. it will return `true` even if the file contents have not changed.

Reply:
It will return true if the file has been deleted by os.Remove(), is that not right?